### PR TITLE
[Alert activation] Add reset_time to alert activation

### PR DESCRIPTION
**Pull request by rokatyy**
Adds new reset_time to alert activation table, so we can determine:

1. whether alert activation is now active or not
2. when alert activation was reset

Jira - https://iguazio.atlassian.net/browse/ML-8432

### CODE REVIEW

File: server/py/framework/db/base.py [hunk 1]
```diff
@@ -968,12 +968,12 @@ def store_alert_activation(
         pass
 
     @abstractmethod
-    def update_alert_activation_number_of_events(
+    def update_alert_activation_on_reset(
```
Comment (TomerShor) [inline server/py/framework/db/base.py]: Despite my comment in the previous PR, let's just call it `update_alert_activation` if we'll maybe use it in other places and update other things.

File: server/py/framework/db/sqldb/db.py [hunk 2]
```diff
@@ -5946,14 +5946,17 @@ def store_alert_activation(
                 session, alert_activation_record, "id"
             )
         else:
+            # for auto reset policy reset_time is the same as the activation time
+            # for manual reset policy, we keep it empty until policy reset
```
Comment (TomerShor) [inline server/py/framework/db/sqldb/db.py]:
```suggestion
            # for manual reset policy, we keep it empty until the alert is reset
```

File: server/py/services/alerts/crud/alerts.py [hunk 3]
```diff
@@ -410,24 +410,23 @@ def reset_alert(self, session: sqlalchemy.orm.Session, project: str, name: str):
...
-                )
-                activation_time = alert_state.get("last_updated")
-                activation_id = alert_state.get("full_object", {}).get(
-                    "last_activation_id"
+            alert_state = framework.utils.singletons.db.get_db().get_alert_state_dict(
+                session, alert.id
+            )
+            activation_time = alert_state.get("last_updated")
+            activation_id = alert_state.get("full_object", {}).get("last_activation_id")
+            if activation_time and activation_id:
```
Comment (TomerShor) [inline server/py/services/alerts/crud/alerts.py:418]: If one or both are None - is it a valid case?
If not, we should at least log a warning.

**Review by TomerShor — CHANGES_REQUESTED** [review]
Minor comments

File: server/py/framework/db/sqldb/db.py (hunk 2 above)
Comment (rokatyy) [reply server/py/framework/db/sqldb/db.py]: @TomerShor actually, that's alert being reset, reset policy just indicates when to reset it

File: server/py/services/alerts/crud/alerts.py (hunk 3 above)
Comment (rokatyy) [reply server/py/services/alerts/crud/alerts.py:418]: @TomerShor nope - not a valid case, will add warning

File: server/py/framework/db/sqldb/db.py (hunk 2 above)
Comment (TomerShor) [reply server/py/framework/db/sqldb/db.py]: exactly, the alert is reset.
"until policy reset" means the the `policy` is reset, which is not the case.

File: server/py/services/alerts/crud/alerts.py [hunk 4]
```diff
@@ -410,24 +410,30 @@ def reset_alert(self, session: sqlalchemy.orm.Session, project: str, name: str):
...
+                    number_of_events=number_of_events
+                    if number_of_events > alert.criteria.count
+                    else None,
                 )
-                activation_time = alert_state.get("last_updated")
-                activation_id = alert_state.get("full_object", {}).get(
-                    "last_activation_id"
+            else:
+                logger.warning(
+                    "No activation id or last activation time found for alert",
```
Comment (TomerShor) [inline server/py/services/alerts/crud/alerts.py]:
```suggestion
                    "No activation id or last activation time found for alert, skipping activation update on reset",
```

File: server/py/framework/db/sqldb/db.py (hunk 2 above)
Comment (rokatyy) [reply server/py/framework/db/sqldb/db.py]: @TomerShor right, sorry :facepalm:

**Review by Yacouby — APPROVED** [review]
Nice 👍

**Review by TomerShor — APPROVED** [review]

**Pull request merged**